expo/expo
# [docs] Upgrade next to 12.1.5 and switch over to swc

# Why

This bumps up Next from 11 to 12, and does a few cleanup things.

# How

1. Remove the ESBuild integration, we would be using SWC now which is fast enough
2. Switch over to Webpack 5, next 12 doesn't support 4 anymore
3. Only add Babel back for the preval mechanism in the **docs/constants/**** files
4. Enable and configure SWC for the docs, including experimental emotion and minifier
5. Add a resolution-workaround for MDX v1 to use only 1 babel version

# Test Plan

- See if it builds in CI
- See if it builds locally with `yarn dev`
- Check if the pages are loading properly, e.g.
  - API version pages
  - API app config page
  - Random guides
  - Random RN API pages

# Checklist
